src/main/java/invtweaks/container/DirectContainerManager.java

```diff
@@ -82,8 +82,8 @@ public boolean move(ContainerSection srcSection, int srcIndex, ContainerSection
 
         // Use intermediate slot if we have to swap tools, maps, etc.
         assert srcStack != null;
-        if(destStack != null && srcStack.getItem() == destStack.getItem() && (srcStack.getMaxStackSize() == 1 ||
-                srcStack.hasTagCompound() || destStack.hasTagCompound())) {
+        if(destStack != null && srcStack.isItemEqual(destStack) && srcStack.getMaxStackSize() == 1 &&
+                ((srcStack.hasTagCompound() || destStack.hasTagCompound()) && (srcStack.getTagCompound().equals(destStack.getTagCompound())))) {
```

Comment:
this tag problem also has notice by the other user that has replied in forum. but I did not know that is the same problem of mine.